module/purpose/Soccer/src/Soccer.cpp

```diff
@@ -121,13 +129,39 @@ namespace module::purpose {
             }
         });
 
+        // Left button pauses the soccer scenario
+        on<Trigger<ButtonLeftDown>, Single>().then([this] {
+            emit<Scope::DIRECT>(std::make_unique<ResetFieldLocalisation>());
+            emit<Scope::DIRECT>(std::make_unique<EnableIdle>());
+            emit<Scope::DIRECT>(std::make_unique<Buzzer>(1000));
+            idle = true;
+        });
+
+        on<Trigger<ButtonLeftUp>, Single>().then([this] { emit<Scope::DIRECT>(std::make_unique<Buzzer>(0)); });
+
+        on<Trigger<EnableIdle>, Single>().then([this] {
+            // Stop all tasks and stand still
+            emit<Task>(std::unique_ptr<FindPurpose>(nullptr));
+            emit(std::make_unique<Stability>(Stability::UNKNOWN));
+            log<NUClear::INFO>("Idle mode enabled");
+        });
+
+        // Middle button resumes the soccer scenario
         on<Trigger<ButtonMiddleDown>, Single>().then([this] {
-            // Middle button forces playing
-            log<NUClear::INFO>("Middle button pressed!");
-            if (!cfg.force_playing) {
-                log<NUClear::INFO>("Force playing started.");
-                cfg.force_playing = true;
+            emit<Scope::DIRECT>(std::make_unique<ResetFieldLocalisation>());
+            // Restart the Director graph for the soccer scenario after a delay
+            emit<Scope::DELAY>(std::make_unique<DisableIdle>(), std::chrono::seconds(cfg.disable_idle_delay));
+            emit<Scope::DIRECT>(std::make_unique<Buzzer>(1000));
+            idle = false;
+        });
+
+        on<Trigger<ButtonMiddleUp>, Single>().then([this] { emit<Scope::DIRECT>(std::make_unique<Buzzer>(0)); });
+
+        on<Trigger<DisableIdle>, Single>().then([this] {
+            // If the robot is not idle, restart the Director graph for the soccer scenario!
+            if (!idle) {
```

Comment:
how come we have this idle variable?

Reply:
Since we have a delay, you can press the disable idle then the enable idle button, but since we have the delay the enable idle won't prevent this trigger from running, so I added flag which is updated instantly